From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LuneOS / webOS Ports <team@webos-ports.org>
Date: Thu, 13 Aug 2026 00:00:00 +0000
Subject: [PATCH] CMakeLists: use ${{VAR}} instead of @VAR@ for variable references

CMake's CMP0053, introduced in 3.1, stopped expanding @VAR@ in unquoted
arguments. Until now this worked only because webOS.cmake declared
cmake_minimum_required(VERSION 2.8.7), which left the policy at OLD. Raising
that to 3.5 for CMake 4 - and CMake 4 removes the OLD behaviour of every policy
older than 3.5, so it cannot be set back - makes @VAR@ a literal string:

  Files/directories were installed but not shipped in any package:
    /usr/@WEBOS_INSTALL_INCLUDEDIR@
    /usr/@WEBOS_INSTALL_INCLUDEDIR@/...

Upstream-Status: Pending
---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7593451..c837c0c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,7 +97,7 @@ webos_include_install_paths()
 webos_build_db8_files()
 
 install(TARGETS ${BIN_NAME} DESTINATION sbin)
-install(FILES @CMAKE_SOURCE_DIR@/files/rules/39-usb-auto-port.rules DESTINATION @WEBOS_INSTALL_SYSCONFDIR@/udev/rules.d/)
-install(DIRECTORY @CMAKE_SOURCE_DIR@/files/rules/ DESTINATION @WEBOS_INSTALL_SYSCONFDIR@/udev/rules.d/ FILES_MATCHING PATTERN "*.rules*" PATTERN ".*" EXCLUDE)
-install(DIRECTORY @CMAKE_SOURCE_DIR@/files/icon/ DESTINATION @WEBOS_INSTALL_DATADIR@/@CMAKE_PROJECT_NAME@ FILES_MATCHING PATTERN "*.png*" PATTERN ".*" EXCLUDE)
-install(DIRECTORY "inc/public/" DESTINATION @WEBOS_INSTALL_INCLUDEDIR@ FILES_MATCHING PATTERN "*.h*" PATTERN ".*" EXCLUDE)
+install(FILES ${CMAKE_SOURCE_DIR}/files/rules/39-usb-auto-port.rules DESTINATION ${WEBOS_INSTALL_SYSCONFDIR}/udev/rules.d/)
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/files/rules/ DESTINATION ${WEBOS_INSTALL_SYSCONFDIR}/udev/rules.d/ FILES_MATCHING PATTERN "*.rules*" PATTERN ".*" EXCLUDE)
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/files/icon/ DESTINATION ${WEBOS_INSTALL_DATADIR}/${CMAKE_PROJECT_NAME} FILES_MATCHING PATTERN "*.png*" PATTERN ".*" EXCLUDE)
+install(DIRECTORY "inc/public/" DESTINATION ${WEBOS_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.h*" PATTERN ".*" EXCLUDE)
